R/translator.R

```diff
@@ -0,0 +1,138 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    key_translation = "character",
+    translation_language = "character",
+    translation_file = "character",
+    options = "list",
+    translations = "data.frame",
+    mode = "character"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_file, key_translation) {
+    translation_file <<- translation_file
+    key_translation <<- key_translation
+    options <<- .translator_options
+    # determine the translations format
+    translation_split <- unlist(strsplit(translation_file, "[.]"))
+    translation_format <- translation_split[length(translation_split)]
+    if (length(translation_split) == 1)
```

Comment:
That's kind of an issue. I was thinking about it for a while.

1) First of all, imho it's worth to support both jsons and csvs reading.
2) Giving only one `tran_en.csv` seems strange and even more confusing for me, as it might make user think that files order has some meaning, whereas it doesn't.
3) That's why I decided for a solution with prefix, but agree that's it's not sufficiently documented yet.

But now is a good moment to discuss the options we have. I would stick to my solution, but maybe @marekrogala has sth to add?

Reply:
why not have separate mutually exclusive arguments for that?:
`initialize("pl", translation_csvs_path = "i18n/translations/")`, `initialize("pl", translation_json_path = "i18n/translations.json")` ?
+ additionally, why not assume that translation files are in some specific path and simply are named like `$LANGUAGE.csv` ?

Reply:
Actually, that I like this idea the most. Thanks!